pallets/confidential-asset/src/lib.rs

```diff
@@ -137,7 +119,7 @@ decl_module! {
             let owner_acc = ensure_signed(origin)?;
             let owner_id = Context::current_identity_or::<Identity<T>>(&owner_acc)?;
 
-            let valid_asset_ids = convert_asset_ids(Self::valid_asset_ids());
+            let valid_asset_ids = convert_asset_ids(Self::confidential_tickers());
```

Comment:
Is anyone allowed to call this dispatchable?

Reply:
do you mean the `confidential_tickers()`? yes, wallet will need to call it eventually.

Reply:
Sorry, I had to specific. I mean the dispatchable `validate_mercat_account()`. Let's say I am a polymesh user and I want to invest in some confidential asset then I will call this function or someone on my behalf?

Reply:
Yes, before your MERCAT account gets added someone has to validate that you are in fact planning to hold an asset that's among confidential assets, and you start from a 0 balance. Since the asset type is hidden from MESH the user will have to call this function to show their account is valid.